extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/DigitalSTROMHandlerFactory.java

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal;
+
+import static org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants;
+import org.eclipse.smarthome.binding.digitalstrom.handler.DeviceHandler;
+import org.eclipse.smarthome.binding.digitalstrom.handler.SceneHandler;
+import org.eclipse.smarthome.binding.digitalstrom.handler.BridgeHandler;
+import org.eclipse.smarthome.binding.digitalstrom.internal.discovery.SceneDiscoveryService;
+import org.eclipse.smarthome.binding.digitalstrom.internal.discovery.DeviceDiscoveryService;
+import org.eclipse.smarthome.binding.digitalstrom.internal.discovery.SceneDiscoveryServiceManager;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.ConnectionManager;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.impl.ConnectionManagerImpl;
+import org.eclipse.smarthome.binding.digitalstrom.internal.providers.DsThingTypeProvider;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingTypeProvider;
+import org.osgi.framework.ServiceRegistration;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link DigitalSTROMHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Mathias Siegele - Initial contribution
+ *
+ */
+public class DigitalSTROMHandlerFactory extends BaseThingHandlerFactory {
+
+    private Map<String, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.union(SceneHandler.SUPPORTED_THING_TYPES,
+            Sets.union(BridgeHandler.SUPPORTED_THING_TYPES, DeviceHandler.SUPPORTED_THING_TYPES));
+
+    private ConnectionManager connMan = null;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES.contains(thingTypeUID);
```

Comment:
Yes, that’s right. If the system was restarted, the DeviceHandler can’t be created for existing dSDevices till the Bridge status turns to online.
But the problem is gone, because we follow your suggestion. But we think it’s better to create  ThingTypes from the supported funktional-color-groups of digitalSTROM GE, SW and GR, like dS-GE-Device for all light devices, like e.g. the GE-KM200, instead of a GenericDSDevice. Because the GE- and SW- devices in contrast to the GR-Devices for shades doesn’t have sensors to read power consumptions and so there are no configurations to set the priorities of sensor types.
We think this is also better for the future, if new devices get introduced by digitalSTROM to implement devices of new color-groups. This color-groups or devices can have also other configurations and so it has to be create only a new ThingType with this configurations for this group or device.